luminoso_api/v5_cli.py

```diff
@@ -68,10 +69,67 @@ def _read_params(input_file, json_body, p_params):
     return params
 
 
+def connect_with_token_args(args):
+    """
+    A shared function for working with Luminoso auth tokens in command line
+    utilities, interactively if necessary.
+    """
+    token = args.token
+    save_token = args.save_token
+    token_error = None
+
+    try:
+        client = LuminosoClient.connect(args.base_url, token=token)
+        # Make an authenticated request to test that the token is valid
+        client.get('/projects')
+    except LuminosoAuthError as e:
+        token_error = e
+
+    if token_error:
+        if token is not None:
+            # We were explicitly given a token that doesn't work, so re-raise
+            # the error
+            raise token_error
+        else:
```

Comment:
If the user has an invalid token stored for the domain, what could they possibly do with it except replace it with a valid one? The token may have simply expired, so I don't think the user has necessarily made a mistake that they need to be alerted about. So this just goes on to the next thing the user would need to do in the workflow, which is to get a token.

Perhaps I'm not understanding what the major problem is.

Reply:
The main thing is that we're not correctly identifying that there is a saved token, it just has a problem, and this may cause the user to think the token didn't get stored at all.  This just strikes me as a bad user interface.